sdk/core/azure-core/azure/core/pipeline/policies/_retry.py

```diff
@@ -317,7 +318,21 @@ def increment(self, settings, response=None, error=None):
                 settings['status'] -= 1
                 settings['history'].append(RequestHistory(response.http_request, http_response=response.http_response))
 
-        return not self.is_exhausted(settings)
+        if not self.is_exhausted(settings):
+            if response.http_request.body and hasattr(response.http_request.body, 'read'):
+                try:
+                    body_position = response.request.http_request.body.tell()
+                except (AttributeError, UnsupportedOperation):
+                # if body position cannot be obtained, then retries will not work
+                    return False
+                try:
+                    # attempt to rewind the body to the initial position
+                    response.http_request.body.seek(body_position, SEEK_SET)
+                except (UnsupportedOperation, ValueError, AttributeError):
+                    # if body is not seekable, then retry would not work
+                    return False
+            return True
+        return False
```

Comment:
If you refactor this inner block into a helper function this could be written:
```
if not self.is_exhausted(settings):
    return response.http_request.body and is_seekable(response.http_request.body)
return False
```
and then also `is_seekable` can get very simple unit tests that cover all the branches (no pipeline or mocking needed)